src/Microsoft.ML.Tokenizers/Microsoft.ML.Tokenizers.csproj

```diff
@@ -32,17 +32,37 @@
       <Files ParameterType="Microsoft.Build.Framework.ITaskItem[]" Required="true" />
     </ParameterGroup>
     <Task>
+      <Using Namespace="System.Globalization" />
       <Using Namespace="System.IO" />
       <Using Namespace="System.IO.Compression" />
       <Code Type="Fragment" Language="cs">
 			<![CDATA[
-        foreach(var file in Files)
+        foreach (var file in Files)
         {
-            using var sourceStream = File.OpenRead(file.GetMetadata("FullPath"));
+            string fileName = file.GetMetadata("FullPath");
+            string fileContent = File.ReadAllText(fileName);
+            int capacity = 1;
+            int eolIndex = 0;
+            do
+            {
+                if ((eolIndex = fileContent.IndexOf('\n', eolIndex)) >= 0)
+                {
+                    eolIndex++;
+                    capacity++;
+                }
+                else
+                {
+                    break;
+                }
+            } while (eolIndex < fileContent.Length);
```

Comment:
I assume this code needs to compile with netstandard and run on netfx?  if it could rely on net8.0, this could just use Count

Reply:
@stephentoub 

I am seeing 9.0 preview SDK is used when building the netstandard 2.0 library. Could you elaborate more on what `Count` API I can use in .NET 8.0? 

```
C:\Program Files\dotnet\dotnet.exe "C:\Program Files\dotnet\sdk\9.0.100-preview.3.24204.13\Roslyn\bincore\csc.dll" /noconfig /reference:"C:\Program Files\dotnet\sdk\9.0.100-preview.3.24204.13\ref\Microsoft.Build.Framework.dll" /reference:"C:\Program Files\dotnet\sdk\9.0.100-preview.3.24204.13\ref\Microsoft.Build.Utilities.Core.dll" /reference:"C:\Program Files\dotnet\sdk\9.0.100-preview.3.24204.13\ref\mscorlib.dll" /reference:"C:\Program Files\dotnet\sdk\9.0.100-preview.3.24204.13\ref\netstandard.dll" /deterministic+ /nologo /optimize- /target:Library /out:C:\Users\tarekms.NTDEV\AppData\Local\Temp\MSBuildTemp\tmp0f4ed34c158f4e61ab301496d80e6299.dll C:\Users\tarekms.NTDEV\AppData\Local\Temp\MSBuildTemp\tmpfd8784f3249541ef89fd8712e4966643.tmp /nostdlib+ 
```

CC @ericstj

Reply:
https://learn.microsoft.com/en-us/dotnet/api/system.memoryextensions.count?view=net-8.0#system-memoryextensions-count-1(system-readonlyspan((-0))-0)

fileContent.AsSpan().Count('\n')

Reply:
Thanks. Unfortunately, it didn't work as the compilation ref netstandard.dll.